### ISSUE TITLE
[Regression] UI stops responding to input after clicking `Reconnect`

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#4509]
##### Brief Description

After repeatedly quickly clicking on `Reconnect` button in Advanced settings qTox's UI stops responding to user input and no longer connects to the network. UI doesn't actually freeze, just doesn't react to input. The only way to close qTox is to kill it.

OS: Gentoo/Linux
qTox version: v1.10.0-114-gd704f5d2
Commit hash: d704f5d21d7b33fbb52f89d9d2a5386015e6a3ce
toxcore: 0.1.8
Qt: 5.6.2

Reproducible: Always

##### Steps to reproduce

1. Turn on `Multiple windows mode`
2. Click on `Reconnect` in settings
3. Keep repeating point 2., do it quickly

##### Observed Behavior
UI stops responding to user input, qTox no longer connects to network

##### Expected Behavior
qTox should just reconnect

##### Additional Info

log:

```
Starting program: /home/zetok/working/qtox-test/qTox/_build/qtox -p _3
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[11:47:05.258 UTC] qxcbintegration.cpp:167 : Debug: Qt: gdb: -nograb added to command-line options.
	 Use the -dograb option to enforce grabbing.
[New Thread 0x7fffdeb79700 (LWP 4878)]
[11:47:05.444 UTC] :0 : Debug: using qt5ct plugin
[New Thread 0x7fffde06b700 (LWP 4929)]
[New Thread 0x7fffd7132700 (LWP 4979)]
[11:47:07.264 UTC] persistence/settings.cpp:133 : Debug: "Loading settings from /home/zetok/.config/tox/qtox.ini"
[11:47:07.297 UTC] ipc.cpp:61 : Debug: Our global IPC ID is  11121830072949958205
[11:47:07.297 UTC] ipc.cpp:73 : Debug: Attaching to the global shared memory
[11:47:07.297 UTC] ipc.cpp:251 : Debug: Previous owner timed out, taking ownership 13506129366572039415 -> 11121830072949958205
[11:47:07.297 UTC] main.cpp:234 : Debug: built on:  12:46:44 Jul  8 2017 ( 1499514376 )
[11:47:07.297 UTC] main.cpp:235 : Debug: commit:  d704f5d21d7b33fbb52f89d9d2a5386015e6a3ce 

[11:47:07.941 UTC] persistence/profile.cpp:110 : Debug: Loading tox save  "/home/zetok/.config/tox/_3.tox"
[11:47:07.941 UTC] persistence/settings.cpp:453 : Debug: "Saving global settings at /home/zetok/.config/tox/qtox.ini"
[11:47:08.086 UTC] persistence/settings.cpp:327 : Debug: Loading personal settings from "/home/zetok/.config/tox/_3.ini"
[11:47:08.090 UTC] nexus.cpp:90 : Debug: Starting up
[New Thread 0x7fffc263f700 (LWP 5128)]
[New Thread 0x7fffc1c39700 (LWP 5129)]
[New Thread 0x7fffc1438700 (LWP 5130)]
[New Thread 0x7fffc0c37700 (LWP 5131)]
[New Thread 0x7fffc0436700 (LWP 5132)]
[11:47:10.158 UTC] kernel/qobject.cpp:4674 : Warning: QObject::connect: invalid null parameter
[New Thread 0x7fffbf07b700 (LWP 5133)]
[New Thread 0x7fffbe87a700 (LWP 5182)]
[New Thread 0x7fffbe079700 (LWP 5183)]
[New Thread 0x7fffaffff700 (LWP 5184)]
[11:47:10.939 UTC] core/core.cpp:264 : Debug: Loading user profile
[11:47:10.939 UTC] core/core.cpp:130 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
[New Thread 0x7fffaf7fe700 (LWP 5185)]
[11:47:10.944 UTC] :0 : Debug: D-Bus system tray: no
[11:47:10.944 UTC] widget/systemtrayicon.cpp:82 : Debug: Using the Qt backend
[New Thread 0x7fffaeffd700 (LWP 5186)]
[11:47:14.453 UTC] core/core.cpp:406 : Debug: "Connecting to 2400:6180:0:d0::17a:a001:33445 (wiiaam)"
[11:47:14.454 UTC] core/core.cpp:406 : Debug: "Connecting to 37.97.185.116:33445 (Yani)"
[11:47:18.953 UTC] core/core.cpp:406 : Debug: "Connecting to 51.254.84.212:33445 (a68366)"
[11:47:18.953 UTC] core/core.cpp:406 : Debug: "Connecting to loki.uplinklabs.net:33445 (AbacusAvenger)"
[11:47:19.198 UTC] core/core.cpp:415 : Debug: "Error bootstrapping from AbacusAvenger"
[11:47:19.349 UTC] core/core.cpp:419 : Debug: "Error adding TCP relay from AbacusAvenger"
[11:47:22.666 UTC] core/core.cpp:374 : Debug: Connected to the DHT
[11:47:22.666 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:30.743 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[Thread 0x7fffaf7fe700 (LWP 5185) exited]
[11:47:31.041 UTC] core/core.cpp:264 : Debug: Loading user profile
[11:47:31.041 UTC] core/core.cpp:130 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
[New Thread 0x7fffaf7fe700 (LWP 6260)]
[New Thread 0x7fffae7fc700 (LWP 6261)]
[Thread 0x7fffaeffd700 (LWP 5186) exited]
[11:47:33.082 UTC] core/core.cpp:377 : Debug: Disconnected from the DHT
[11:47:34.553 UTC] core/core.cpp:406 : Debug: "Connecting to tox.ntp.moscow:33445 (abbat)"
[11:47:34.756 UTC] core/core.cpp:415 : Debug: "Error bootstrapping from abbat"
[11:47:34.970 UTC] core/core.cpp:419 : Debug: "Error adding TCP relay from abbat"
[11:47:34.970 UTC] core/core.cpp:406 : Debug: "Connecting to bootybay.club:33445 (AssPirate)"
[11:47:35.010 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[Thread 0x7fffaf7fe700 (LWP 6260) exited]
[11:47:37.219 UTC] core/core.cpp:264 : Debug: Loading user profile
[11:47:37.219 UTC] core/core.cpp:130 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
[New Thread 0x7fffaf7fe700 (LWP 6572)]
[New Thread 0x7fffaeffd700 (LWP 6573)]
[Thread 0x7fffae7fc700 (LWP 6261) exited]
[Thread 0x7fffbf07b700 (LWP 5133) exited]
[11:47:41.096 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[Thread 0x7fffaf7fe700 (LWP 6572) exited]
[11:47:41.417 UTC] core/core.cpp:264 : Debug: Loading user profile
[11:47:41.417 UTC] core/core.cpp:130 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
[New Thread 0x7fffaf7fe700 (LWP 6820)]
[New Thread 0x7fffbf07b700 (LWP 6821)]
[Thread 0x7fffaeffd700 (LWP 6573) exited]
[11:47:44.153 UTC] core/core.cpp:406 : Debug: "Connecting to tox.neverlocate.me:33445 (brandon)"
[11:47:50.288 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:50.523 UTC] core/core.cpp:415 : Debug: "Error bootstrapping from brandon"
[11:47:50.708 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:50.946 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.107 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.267 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.427 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.686 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.757 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.905 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:51.991 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.080 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.169 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.255 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.336 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.410 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.491 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.579 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.657 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.846 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:52.925 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.027 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.102 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.191 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.309 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.509 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.787 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.868 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:53.967 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.057 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.134 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.212 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.428 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.502 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.581 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.668 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.757 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.836 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:54.915 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.002 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.080 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.168 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.246 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.335 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.414 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.488 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.570 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.643 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.725 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.833 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:55.927 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.010 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.092 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.177 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.270 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.345 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.426 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.499 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.581 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.655 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.736 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.799 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.881 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.955 UTC] persistence/profile.cpp:322 : Debug: Saving tox save to  "/home/zetok/.config/tox/_3.tox"
[11:47:56.996 UTC] core/core.cpp:419 : Debug: "Error adding TCP relay from brandon"
[11:47:56.996 UTC] core/core.cpp:406 : Debug: "Connecting to 198.98.51.198:33445 (Busindre)"
[Thread 0x7fffaf7fe700 (LWP 6820) exited]
[11:47:57.189 UTC] core/core.cpp:264 : Debug: Loading user profile
[11:47:57.190 UTC] core/core.cpp:130 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
[New Thread 0x7fffaf7fe700 (LWP 7575)]
[New Thread 0x7fffaeffd700 (LWP 7576)]
[Thread 0x7fffbf07b700 (LWP 6821) exited]
[Thread 0x7fffaf7fe700 (LWP 7575) exited]
```

**Comment by Diadlo** [comment]
Need to disable button until core restart

**Comment by ghost** [comment]
@Diadlo 
Hello. I'm disable button until core restart in AdvancedForm::on_reconnectButton_clicked. This move not fixed bug.

**Comment by anthonybilinski** [comment]
Fixed by #5778, present in nightlies. Closing issue.
